src/gui/map/map_editor.cpp

```diff
@@ -3901,6 +3911,318 @@ void MapEditorController::createSymbolWidget(QWidget* parent)
+	{
+		QMessageBox::warning(window, tr("Error Exporting"), tr("Failed to properly create the export georeferencing info"));
+		return;
+	}
+
+	// Make sure an object is selected
+	if (map->getNumSelectedObjects() == 0) {
+		QMessageBox::warning(window, tr("Nothing selected"), tr("No objects are selected.  Not exporting anything."));
+		return;
+	}
+
+	QSettings settings;
+	QString export_directory = settings.value(QString::fromLatin1("exportFileDirectory"), QDir::homePath()).toString();
+
+	QString filename = QFileDialog::getSaveFileName(
+						   window,
+						   tr("Export GPX, KML, or SHP file"),
+						   export_directory,
+						   tr("GPS Exchange Format (*.gpx);;Keyhole Markup Language (*.kml);;Shapefile (*.shp)"));
+	if (filename.isEmpty() || filename.isNull())
+		return;
+
+	settings.setValue(QString::fromLatin1("exportFileDirectory"), QFileInfo(filename).canonicalPath());
+
+	// Setup format-specific features
+	const char *psz_driver_name;
+	bool gpx_quirks = false;
+	if (filename.endsWith(QLatin1String(".gpx"), Qt::CaseInsensitive))
+	{
+		psz_driver_name = "GPX";
+		gpx_quirks = true;
+	}
+	else if (filename.endsWith(QLatin1String(".kml"), Qt::CaseInsensitive))
+	{
+		psz_driver_name = "KML";
+	}
+	else
+		psz_driver_name = "ESRI Shapefile";
+
+	// GPX driver can't automatically overwrite files
+	if (gpx_quirks)
+	{
+		QFile file(filename);
+		if (file.exists())
+			file.remove();
+	}
+
+	GDALAllRegister();
+
+	auto po_driver = GetGDALDriverManager()->GetDriverByName(psz_driver_name);
+	if (po_driver == nullptr)
+	{
+		QMessageBox::warning(window,
+							 tr("Error Exporting"),
+							 tr("Couldn't find required GDAL driver %1.  Not exporting.").arg(QString::fromLatin1(psz_driver_name)));
+		return;
+	}
+
+	// Create output dataset
+	char **papsz_options = nullptr;
+	papsz_options = CSLSetNameValue(papsz_options, "GPX_USE_EXTENSIONS", "YES");
+	papsz_options = CSLSetNameValue(papsz_options, "GDAL_VALIDATE_CREATION_OPTIONS", "NO");
+	auto po_ds = po_driver->Create(filename.toLatin1(), 0, 0, 0, GDT_Unknown, papsz_options);
+	if (po_ds == nullptr)
+	{
+		QMessageBox::warning(window,
+							 tr("Error Exporting"),
+							 tr("Failed to create output file.  Do you have write permissions in this folder?"));
+		CSLDestroy(papsz_options);
+		return;
+	}
+
+	/**
+	 * Create collections of point, line, and areas that are selected
+	 * These have to be separate as some drivers required them to be added in a specific order
+	 */
+	std::vector<const Object*> point_objects;
+	std::vector<const Object*> line_objects;
+	std::vector<const Object*> area_objects;
+	for (const auto& object: map->selectedObjects())
+	{
+		if (object->getSymbol()->getType() == Symbol::Point)
+			point_objects.push_back(object);
+		else if (object->getSymbol()->getType() == Symbol::Line)
+			line_objects.push_back(object);
+		else if (object->getSymbol()->getType() == Symbol::Area)
+			area_objects.push_back(object);
+	}
```

Comment:
No text objects. 
Alternative to copying, you might iterate over the same selection multiple times.